crates/aptos-dkg/tests/barycentric_eval.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use aptos_dkg::algebra::polynomials::barycentric_eval;
+use ark_ec::{pairing::Pairing, AdditiveGroup};
+use ark_ff::{FftField, Field, UniformRand};
+use ark_poly::{univariate::DensePolynomial, DenseUVPolynomial, Polynomial};
+use ark_std::{rand::Rng, test_rng};
+
+#[cfg(test)]
+fn run_barycentric_case<E: Pairing>(degree: usize, n: usize, sample_points: usize) {
+    let mut rng = test_rng();
+    type Fr<E> = <E as Pairing>::ScalarField;
+
+    // Generate coefficients for a random polynomial; if degree is 0, randomly sometimes replace with zero
+    let poly_coeffs: Vec<Fr<E>> = if degree == 0 && rng.gen_bool(0.5) {
+        vec![Fr::<E>::ZERO]
+    } else {
+        (0..=degree).map(|_| Fr::<E>::rand(&mut rng)).collect()
+    };
+    let poly = DensePolynomial::from_coefficients_vec(poly_coeffs);
+
+    // Build domain and evaluations
+    let omega = Fr::<E>::get_root_of_unity(n as u64)
+        .unwrap_or_else(|| panic!("no root of unity of size {} for this field", n));
+    let roots: Vec<Fr<E>> = (0..n).map(|i| omega.pow([i as u64])).collect();
+    let evals: Vec<Fr<E>> = roots.iter().map(|&r| poly.evaluate(&r)).collect();
+    let n_inv = Fr::<E>::from(n as u64).inverse().unwrap();
+
+    // Random points
+    for _ in 0..sample_points {
+        let x = Fr::<E>::rand(&mut rng);
+        let expected = poly.evaluate(&x);
+        let val = barycentric_eval(&evals, &roots, x, n_inv);
+        assert_eq!(
+            val, expected,
+            "Failed for degree {}, n = {} at x = {:?}",
+            degree, n, x
+        );
+    }
+
+    // Interpolation points (roots)
+    for (root, &eval) in roots.iter().zip(evals.iter()) {
+        let val = barycentric_eval(&evals, &roots, *root, n_inv);
```

Comment:
This is testing the trivial case where one uses (although they need not to) barycentric interpolation to evaluate at a root of unity, right?

Reply:
Don't remove it cause it's a good thing to test for.